src/TimeSpan.php

```diff
@@ -39,6 +39,37 @@ public static function from(
         );
     }
 
+    public static function fromInterval(\DateInterval $interval): self
+    {
+        if ($interval->m || $interval->y) {
+            throw new \InvalidArgumentException(
+                \sprintf(
+                    'Month and year cannot be converted to microseconds correctly. Use `%s::diff()` instead.',
+                    self::class,
+                ),
+            );
+        }
+
+        if ($interval->days !== false) {
+            throw new \InvalidArgumentException(
+                \sprintf(
+                    'Detected `%s::diff()`. Use `%s::diff()` instead.',
```

Comment:
Let's discuss this later. Some links to consider:
- https://www.php.net/manual/en/datetime.diff.php#refsect1-datetime.diff-returnvalues
- https://en.wikipedia.org/wiki/Daylight_saving_time
- https://www.strangebuzz.com/en/snippets/comparing-two-dateinterval-php-objects